phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -1137,7 +1181,7 @@ public MutationState createIndex(CreateIndexStatement statement, byte[][] splits
                     if (hbaseVersion < PhoenixDatabaseMetaData.MUTABLE_SI_VERSION_THRESHOLD) {
                         throw new SQLExceptionInfo.Builder(SQLExceptionCode.NO_MUTABLE_INDEXES).setTableName(indexTableName.getTableName()).build().buildException();
                     }
-                    if (connection.getQueryServices().hasInvalidIndexConfiguration()) {
+                    if (connection.getQueryServices().hasInvalidIndexConfiguration() && !dataTable.isTransactional()) {
```

Comment:
I think hasInvalidIndexConfiguration() should be renamed to something like isMutableIndexWALCodecInstalled(). The actual check happens on the server side. Renaming this way makes it more explicit that the change has to do with mutable secondary indexes and that if the table is transactional, then we don't care about having the right codec in place on region servers.